### DISCUSSION TITLE
Why is from_bcs::from_bytes restricted?

### DISCUSSION THREAD

**Discussion by lumos42** [aptos-labs/aptos-developer-discussions#411] [Questions] [answered]
What are the reasons for keeping from_bcs::from_bytes function private?

**Comment by 0xbe1** [comment, marked as answer]
> if it is public, one can literally create a `Coin<T>` out of thin air. therefore from_bcs::from_bytes needs a lot of safety checks. for now the team decided to just keep it private, may work on that later.

from david.apt
